src/hooks/useUpdateSetting.jsx

```diff
@@ -7,19 +7,24 @@ import { useUserStore } from "@/store/useUserStore";
 const useUpdateSetting = (type) => {
   const { settings, setUserSettings } = useUserStore();
 
-  const updateSetting = async () => {
+  const updateSetting = async (value) => {
     const userId = localStorage.getItem("userId");
     if (!userId) {
       return;
     }
 
-    const updatedSettings = {
-      ...settings,
-      [type]: !settings[type],
-    };
-
-    if (type === SETTING_TYPES.AD_DETECT && !updatedSettings.isAdDetect) {
-      updatedSettings.isReturnChannel = false;
+    const updatedSettings = { ...settings };
+
+    if (type === SETTING_TYPES.AD_REDIRECT_CHANNEL) {
+      updatedSettings[type] = value;
+    } else {
+      updatedSettings[type] = !settings[type];
+      if (
+        type === SETTING_TYPES.AD_DETECT &&
+        !updatedSettings[SETTING_TYPES.AD_DETECT]
+      ) {
+        updatedSettings[SETTING_TYPES.RETURN_CHANNEL] = false;
+      }
```

Comment:
```suggestion
      const toggledValue = !settings[type];
      updatedSettings[type] = toggledValue;
      
      if (type === SETTING_TYPES.AD_DETECT && !toggledValue) {
        updatedSettings[SETTING_TYPES.RETURN_CHANNEL] = false;
      }
```
`updatedSettings[type] = !settings[type];` 이후 바로 `updatedSettings` 값을 조건문에서 사용하는 흐름이 잘 읽히지 않는 것 같아요!

if문의 조건을 `settings`를 기준으로 판단하는 건 어떨까요?
`!settings[type]`를 `toggledValue` 같은 변수로 분리하면 가독성이 더 좋아질 것 같아요!

두 분의 의견2이 궁금합니다!
@Eun0713 @devjaeyoon

Reply:
`toggledValue`로 분리한 쪽이 흐름도 더 자연스럽고, 의도도 잘 나타내는 것 같아요.
개인적으로 희주님이 말씀하신 방향이 읽는 입장에서 이해하기에 조금 더 수월할 거 같다고 생각합니다.

Reply:
두 분 모두 좋은 의견 감사드립니다.

```suggestion
    const isToggleType = type !== SETTING_TYPES.AD_REDIRECT_CHANNEL;
    const toggledValue = isToggleType ? !settings[type] : value;

    const updatedSettings = {
      [type]: toggledValue,
    };

    if (type === SETTING_TYPES.AD_DETECT && !toggledValue) {
      updatedSettings[SETTING_TYPES.RETURN_CHANNEL] = false;
    }
```

희주님의 `useUserStore.js` 리뷰를 반영하고 두 분의 의견을 종합하여 이렇게 만들어봤는데 어떠신가요? @Eun0713 @faber-h

Reply:
`isToggleType`이랑 `toggledValue`로 분리한 부분 정말 좋습니다!
고생하셨습니다!

Reply:
코드 흐름이 더 명확해진 것 같아요~
수고하셨습니다.